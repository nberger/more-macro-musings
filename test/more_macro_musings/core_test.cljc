(ns more-macro-musings.core-test
  (:require #?(:clj  [clojure.test :refer [deftest is]]
               :cljs [cljs.test :refer-macros [deftest is]])
            [more-macro-musings.core #?(:clj :refer :cljs :refer-macros) [given]]))

(deftest given-macro-test
  (given {:foo "bar" :woo "baz"}
    :foo := "bar"
    :woo :!= "bar")
  (given 1
    inc := 2
    dec := 0
    dec :!= 1))
