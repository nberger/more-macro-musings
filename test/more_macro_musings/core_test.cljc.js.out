;;; target/more_macro_musings/core_test.js
more_macro_musings.core_test.given_macro_test.cljs$lang$test = (function (){
try{var values__6135__auto___7658 = cljs.core._conj.call(null,cljs.core._conj.call(null,cljs.core.List.EMPTY,"bar"),new cljs.core.Keyword(null,"foo","foo",1268894036).cljs$core$IFn$_invoke$arity$1(new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"foo","foo",1268894036),"bar",new cljs.core.Keyword(null,"woo","woo",-5639190),"baz"], null)));
var result__6136__auto___7659 = cljs.core.apply.call(null,cljs.core._EQ_,values__6135__auto___7658);
if(cljs.core.truth_(result__6136__auto___7659)){
cljs.test.do_report.call(null,new cljs.core.PersistentArrayMap(null, 4, [new cljs.core.Keyword(null,"type","type",1174270348),new cljs.core.Keyword(null,"pass","pass",1574159993),new cljs.core.Keyword(null,"expected","expected",1583670997),cljs.core.list(new cljs.core.Symbol("clojure.core","=","clojure.core/=",-1788080406,null),cljs.core.list(new cljs.core.Keyword(null,"foo","foo",1268894036),new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"foo","foo",1268894036),"bar",new cljs.core.Keyword(null,"woo","woo",-5639190),"baz"], null)),"bar"),new cljs.core.Keyword(null,"actual","actual",107306363),cljs.core.cons.call(null,cljs.core._EQ_,values__6135__auto___7658),new cljs.core.Keyword(null,"message","message",-406056002),null], null));
} else {
cljs.test.do_report.call(null,new cljs.core.PersistentArrayMap(null, 4, [new cljs.core.Keyword(null,"type","type",1174270348),new cljs.core.Keyword(null,"fail","fail",1706214930),new cljs.core.Keyword(null,"expected","expected",1583670997),cljs.core.list(new cljs.core.Symbol("clojure.core","=","clojure.core/=",-1788080406,null),cljs.core.list(new cljs.core.Keyword(null,"foo","foo",1268894036),new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"foo","foo",1268894036),"bar",new cljs.core.Keyword(null,"woo","woo",-5639190),"baz"], null)),"bar"),new cljs.core.Keyword(null,"actual","actual",107306363),cljs.core._conj.call(null,cljs.core._conj.call(null,cljs.core.List.EMPTY,cljs.core.cons.call(null,new cljs.core.Symbol("clojure.core","=","clojure.core/=",-1788080406,null),values__6135__auto___7658)),new cljs.core.Symbol(null,"not","not",1044554643,null)),new cljs.core.Keyword(null,"message","message",-406056002),null], null));
}

}catch (e7653){var t__6173__auto___7660 = e7653;
cljs.test.do_report.call(null,new cljs.core.PersistentArrayMap(null, 4, [new cljs.core.Keyword(null,"type","type",1174270348),new cljs.core.Keyword(null,"error","error",-978969032),new cljs.core.Keyword(null,"expected","expected",1583670997),cljs.core.list(new cljs.core.Symbol("clojure.core","=","clojure.core/=",-1788080406,null),cljs.core.list(new cljs.core.Keyword(null,"foo","foo",1268894036),new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"foo","foo",1268894036),"bar",new cljs.core.Keyword(null,"woo","woo",-5639190),"baz"], null)),"bar"),new cljs.core.Keyword(null,"actual","actual",107306363),t__6173__auto___7660,new cljs.core.Keyword(null,"message","message",-406056002),null], null));
}
...




;;; Compilation warnings

;;; NO COMPILATION WARNINGS!


;;; Tests run error:

Testing more-macro-musings.core-test

Ran 1 tests containing 5 assertions.
0 failures, 0 errors.
